### DISCUSSION TITLE
2023-5-13 online discussion

### DISCUSSION THREAD

**Discussion by jzhang533** [PFCCLab/paddlefx#44] [General]
[中国软件开源创新大赛：飞桨框架任务挑战赛-paddlefx 赛题](https://github.com/PaddlePaddle/Paddle/issues/53172#paddlepaddle01)的交流会

欢迎大家在会前，回复这个帖子提出想要讨论和了解的问题。

**Comment by zrr1999** [comment]
### 关于forward中self的处理问题

类似下面pr里提到的情况
https://github.com/PFCCLab/paddlefx/pull/43
```python
def forward(self, a, b):
      c = self.fc[0](a)
      d = self.fc[1](b)
```
pytorch（dynamo）里会把 self.fc[0] 变成类似这种 call_module self__fc_1

我之前提交的[pr](https://github.com/PFCCLab/paddlefx/pull/41)在self处理上应该还是有比较多问题的，直接将self作为Proxy会遇到很多问题，
例如上述代码`self.fc[0]`最终会被变成一个call_function，且fn是一个Node，https://github.com/PFCCLab/paddlefx/pull/43 给出了一个临时的解决办法，使codegen是正常的（至少看起来）。
